gradle-palantir-java-format/src/test/groovy/com/palantir/javaformat/gradle/PalantirJavaFormatSpotlessPluginTest.groovy

```diff
@@ -16,30 +16,74 @@
 package com.palantir.javaformat.gradle
 
 import nebula.test.IntegrationTestKitSpec
+import nebula.test.functional.GradleRunner
+import nebula.test.functional.internal.classpath.ClasspathAddingInitScriptBuilder
 import spock.lang.Unroll
 
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+import java.util.stream.Collectors
+import java.util.stream.Stream
+
 class PalantirJavaFormatSpotlessPluginTest extends IntegrationTestKitSpec {
     /** ./gradlew writeImplClasspath generates this file. */
     private static final CLASSPATH_FILE = new File("build/impl.classpath").absolutePath
     private static final NATIVE_IMAGE_FILE = new File("build/nativeImage.path")
     private static final NATIVE_CONFIG = String.format("palantirJavaFormatNative files(\"%s\")", NATIVE_IMAGE_FILE.text)
+    private static final String INIT_FILE_NAME = "init.gradle";
 
 
     @Unroll
-    def "formats with spotless when spotless is applied"(String extraGradleProperties, String expectedOutput) {
+    def "formats with spotless when spotless is applied"(String extraGradleProperties, String javaVersion, String expectedOutput) {
+        File initScript = new File(projectDir, INIT_FILE_NAME)
+        ClasspathAddingInitScriptBuilder.build(initScript, getBuildPluginClasspathInjector().toList())
```

Comment:
I think this should move into `runGradlewTasks` then you don't even need to have a constant. It's kinda similar to how the nebula thing does it, doesn't necessarily have to be a temp file, but I don't think it's anything specific to this particular test unlike `CLASSPATH_FILE` and `NATIVE_CONFIG` etc.